pkg/checkpoint/checkpointvaliditychecker/conservativecvc.go

```diff
@@ -0,0 +1,64 @@
+package checkpointvaliditychecker
+
+import (
+	es "github.com/go-errors/errors"
+
+	"github.com/filecoin-project/mir/pkg/checkpoint"
+	"github.com/filecoin-project/mir/pkg/crypto"
+	issconfig "github.com/filecoin-project/mir/pkg/iss/config"
+	checkpointpbtypes "github.com/filecoin-project/mir/pkg/pb/checkpointpb/types"
+	trantorpbtypes "github.com/filecoin-project/mir/pkg/pb/trantorpb/types"
+	tt "github.com/filecoin-project/mir/pkg/trantor/types"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+type ConservativeCVC struct {
+	issParams    *issconfig.ModuleParams
+	ownID        t.NodeID
+	hashImpl     crypto.HashImpl
+	chkpVerifier checkpoint.Verifier
+}
+
+// NewConservativeCVC returns a new ConservativeCVC. This checkpoint validity checker simply rejects checkpoints whose signatures cannot be verified because
+// the node does not know the membership of the relevant epoch yet.
+func NewConservativeCVC(issParams *issconfig.ModuleParams, ownID t.NodeID, hashImpl crypto.HashImpl, chkpVerifier checkpoint.Verifier) *ConservativeCVC {
```

Comment:
We only ever use `configOffset` from the ISS params, and the CVC is intended not to be part of ISS. And, in fact, we don't even need the `configOffset` value, since it can be derived from `len(memberships)` passed to `Verify`. I thus propose not using the `issconfig.ModuleParams` at all in the CVC. (But watch out for a potential +1 or -1 in with `len(memberships)`, as it contains the current membership as well - thus one more than `configOffset`.)

Reply:
yes the -1 is part of the calculation in the verify. I will directly pass configOffset to the CVC.